src/sentry/api/serializers/models/environment.py

```diff
@@ -1,28 +1,52 @@
 from collections import namedtuple
 from datetime import timedelta
+from typing import List
 
 from django.utils import timezone
 
 from sentry.api.serializers import Serializer, register
 from sentry.app import tsdb
-from sentry.models import Environment, EnvironmentProject
+from sentry.models import Environment, EnvironmentBookmark, EnvironmentProject, User
 
 StatsPeriod = namedtuple("StatsPeriod", ("segments", "interval"))
 
 
+def get_bookmarks(env_list: List[Environment], user: User):
+    """
+    Gets a list of environment id's which are bookmarked by the user
+    """
+    if not user.is_authenticated or not env_list:
+        return set()
+
+    return set(
+        EnvironmentBookmark.objects.filter(
+            user=user, environment_id__in=[i.id for i in env_list]
+        ).values_list("environment_id", flat=True)
+    )
+
+
 @register(Environment)
 class EnvironmentSerializer(Serializer):
+    def get_attrs(self, item_list, user):
+        bookmarks = get_bookmarks(item_list, user)
+        return {item: {"is_bookmarked": item.id in bookmarks} for item in item_list}
```

Comment:
Can you make this an `expand` parameter? No need to fetch this extra info in most cases

Reply:
I think it's likely we _do_ want the bookmark status commonly for these serializers. Since the main place these are used is in the project environment settings page, where we'll show if the environment is stared

Reply:
Actually maybe different for this one, let me verify

Reply:
These can be used in random places, like maybe we serialize all `Environments` associated with a project. It just makes sense to not include extra data here by default.